alloc/src/chain.rs

```diff
@@ -239,6 +239,7 @@ unsafe impl<A: Allocator + Clone> Allocator for ChainAllocator<A> {
         debug_assert!(chain_node.remaining_capacity() >= layout.size());
 
         let result = chain_node.linear.allocate(layout);
+        #[cfg(not(fuzzing))]
         // If this fails, there's a bug in the allocator.
         debug_assert!(result.is_ok());
```

Comment:
I've been thinking about this more, and I think the comment is accurate. The issue is the capacity checks that the chain allocator does do not account for alignment, but the linear allocator does. So the chain thinks there is capacity, and the linear says there isn't. There's the problem.
Of course, for our string table, this is always align=1 so there's no need for alarm. But maybe we should make a method for the linear allocator that is like `has_capacity_for(layout: Layout)` that the chain allocator can call?